inf-clojure.el

```diff
@@ -700,29 +661,52 @@ Fallback to `default-directory.' if not within a project."
                            inf-clojure-project-root-files)))
       default-directory))
 
-(defun inf-clojure-project-type ()
-  "Determine the type, either leiningen or boot of the current project."
-  (or inf-clojure-project-type
-      (let ((default-directory (inf-clojure-project-root)))
-        (cond ((file-exists-p "project.clj") "lein")
-              ((file-exists-p "build.boot") "boot")
-              ((file-exists-p "deps.edn") "tools.deps")
-              (t "generic")))))
-
-(defun inf-clojure-cmd (project-type)
-  "Determine the command `inf-clojure' needs to invoke for the PROJECT-TYPE."
-  (pcase project-type
-    ("lein" inf-clojure-lein-cmd)
-    ("boot" inf-clojure-boot-cmd)
-    ("tools.deps" inf-clojure-tools-deps-cmd)
-    (_ inf-clojure-generic-cmd)))
-
 (defun inf-clojure-clear-repl-buffer ()
   "Clear the REPL buffer."
   (interactive)
   (let ((comint-buffer-maximum-size 0))
     (comint-truncate-buffer)))
 
+(defun inf-clojure-switch-to-repl (eob-p)
+  "Switch to the inferior Clojure process buffer.
+With prefix argument EOB-P, positions cursor at end of buffer."
+  (interactive "P")
+  (if (get-buffer-process inf-clojure-buffer)
+      (let ((pop-up-frames
+             ;; Be willing to use another frame
+             ;; that already has the window in it.
+             (or pop-up-frames
+                 (get-buffer-window inf-clojure-buffer t))))
+        (pop-to-buffer inf-clojure-buffer))
+    (call-interactively #'inf-clojure))
+  (when eob-p
+    (push-mark)
+    (goto-char (point-max))))
+
+(defun inf-clojure-quit (&optional buffer)
```

Comment:
Seems to me command like this one should also check if it was invoked in a REPL buffer and kill it directly.